docs/graphs/graph_algorithms.md

```diff
@@ -0,0 +1,1158 @@
+## Breadth-First-Search (BFS)
+
+Breadth First Search (BFS) is a fundamental graph traversal algorithm. It begins with a node, then first traverses all its adjacent. Once all adjacent are visited, then their adjacent are traversed.
+
+Steps for BFS of a Graph : 
+The algorithm starts from a given source and explores all reachable vertices from the given source. Graphs may contain cycles, so we may come to the same node again. To avoid processing a node more than once, we use a boolean visited array.
+1) While the queue is not empty:
+     Remove a node from the queue and visit it.
+     For each unvisited neighbor of the removed node:
+          Add the neighbor into the queue.
+          Mark the neighbor as visited.
+2) When the queue is empty the algorithm terminates.
+
+Implementation in Java:
+
+Input parameters- 1) ArrayList<ArrayList<Integer>> adj - Adjacency list representation of the graph.
```

Comment:
you need to replace `1) ArrayList<ArrayList<Integer>>` with `1) ArrayList&lt;ArrayList&lt;Integer&gt;&gt;` for solved issues because our code follows mdx with react so, you can't use HTML tags direct.

Reply:
It is resolved. Please review once again.